Sources/Document/CRDT/CRDTTree.swift

```diff
@@ -383,132 +427,83 @@ class CRDTTree: CRDTGCElement {
-        guard let pos = try? self.indexTree.findTreePos(index, true) else {
-            return nil
-        }
-
-        return self.indexTree.findPostorderRight(pos)
-    }
-
-    /**
-     * `findTreePos` finds `TreePos` of the given `CRDTTreePos`
-     */
-    func findTreePos(_ pos: CRDTTreePos, _ editedAt: TimeTicket) throws -> (TreePos<CRDTTreeNode>, CRDTTreeNode) {
-        guard let treePos = self.toTreePos(pos) else {
-            throw YorkieError.unexpected(message: "cannot find node at \(pos)")
-        }
-
-        // Find the appropriate position. This logic is similar to the logical to
-        // handle the same position insertion of RGA.
-        var current = treePos
-        while current.node.next?.pos.createdAt.after(editedAt) ?? false, current.node.parent === current.node.next?.parent {
-            current = TreePos(node: current.node.next!, offset: Int32(current.node.next!.size))
+            self.nodeMapByID.put(node.id, node)
         }
-
-        let right = self.indexTree.findPostorderRight(treePos)
-        return (current, right)
     }
 
     /**
-     * `findTreePosWithSplitText` finds `TreePos` of the given `CRDTTreePos` and
+     * `findNodesAndSplitText` finds `TreePos` of the given `CRDTTreeNodeID` and
      * splits the text node if necessary.
      *
-     * `CRDTTreePos` is a position in the CRDT perspective. This is
+     * `CRDTTreeNodeID` is a position in the CRDT perspective. This is
      * different from `TreePos` which is a position of the tree in the local
      * perspective.
      */
-    func findTreePosWithSplitText(_ pos: CRDTTreePos, _ editedAt: TimeTicket) throws -> (TreePos<CRDTTreeNode>, CRDTTreeNode) {
-        guard let treePos = self.toTreePos(pos) else {
+    func findNodesAndSplitText(_ pos: CRDTTreePos, _ editedAt: TimeTicket) throws -> (CRDTTreeNode, CRDTTreeNode) {
+        guard let treeNodes = self.toTreeNodes(pos) else {
             throw YorkieError.unexpected(message: "cannot find node at \(pos)")
         }
 
+        let parentNode = treeNodes.0
+        var leftSiblingNode = treeNodes.1
+
         // Find the appropriate position. This logic is similar to the logical to
         // handle the same position insertion of RGA.
-        var current = treePos
-        while current.node.next?.pos.createdAt.after(editedAt) ?? false, current.node.parent === current.node.next?.parent {
-            current = TreePos(node: current.node.next!, offset: Int32(current.node.next!.size))
-        }
 
-        if current.node.isText {
-            let split = try current.node.split(current.offset)
+        if leftSiblingNode.isText {
+            let absOffset = leftSiblingNode.id.offset
+            let split = try leftSiblingNode.split(pos.leftSiblingID.offset - absOffset, absOffset)
             if split != nil {
-                self.insertAfter(current.node, split!)
-                split?.insPrev = current.node
+                split!.insPrev = leftSiblingNode
```

Comment:
Can't we just use optional binding instead of force unwrapping optionals?
```swift
if let split = try leftSiblingNode.split(pos.leftSiblingID.offset - absOffset, absOffset)
```

Reply:
Even if we apply our poring policy(following JS code style), we'd better revise this. Thanks.